core/src/test/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -312,6 +312,25 @@ public static double randomDouble() {
         return random().nextDouble();
     }
 
+    /**
+     * Returns a double value in the interval [start, end) if lowerInclusive is set to true,
+     * (start, end) otherwise.
+     *
+     * @param start lower bound of interval to draw uniformly distributed random numbers from
+     * @param end upper bound
+     * @param lowerInclusive whether or not to include lower end of the interval
+     * */
+    public static double randomDoubleBetween(double start, double end, boolean lowerInclusive) {
```

Comment:
Just as an afterthought: GeoDataGenerator is really just a test helper class, so I don't think that its a big problem that it uses static methods form ESTestCase, but if that's the main reason for your concern we can change this of course. It will still uses other randomXY() methods from the same class though.